src/Feature/News/code/Repositories/SearchSettingsRepository.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace Sitecore.Feature.News.Repositories
+{
+  using Sitecore.Data;
+
+  public class SearchSettingsRepository : ISearchSettingsRepository
```

Comment:
Why is this repo in News. Couldn't it be a virtual class in Indexing? This architecture would require each module which has to send ISearchSettings to have its own repository and SearchSettings implementation.